Sources/StreamChat/AudioPlayer/AudioPlaying.swift

```diff
@@ -0,0 +1,375 @@
+    func updateRate(_ newRate: AudioPlaybackRate)
+
+    /// Performs a seek in the loaded asset's timeline at the provided time.
+    /// - Parameter time: The time to seek at
+    func seek(to time: TimeInterval)
+}
+
+/// An implementation of ``AudioPlaying`` that can be used to stream audio files from a URL
+open class StreamRemoteAudioPlayer: AudioPlaying {
+    // MARK: - Properties
+
+    /// The queue used to ensure thread-safe access to the context property
+    private lazy var contextAccessQueue: DispatchQueue = .init(
+        label: "com.getstream.audio.player.context", qos: .userInteractive
+    )
+    /// Describes the player's current playback state. The access to this property is **not** thread-safe
+    private var _context: AudioPlaybackContext = .notLoaded
+    /// Describes the player's current playback state. The access to this property is thread-safe
+    private(set) var context: AudioPlaybackContext {
+        get { contextAccessQueue.sync { _context } }
+        set { contextAccessQueue.sync { _context = newValue } }
+    }
+
+    /// The player that will be used for the playback of the audio files
+    private let player: AVPlayer
+
+    /// The assetPropertyLoader is being used during the loading of an asset with non-nil URL, to provide
+    /// async information about the asset's properties. Currently, we are only loading the `duration`
+    /// property.
+    private let assetPropertyLoader: AssetPropertyLoading
+
+    /// An observer that acts as a mediator between the AVPlayer and StreamAudioPlayer for playback
+    /// updates.
+    private let playerObserver: AudioPlayerObserving
+
+    /// The delegate which should get informed when the player's context gets updated
+    private(set) weak var delegate: AudioPlayingDelegate? {
+        didSet { delegate?.audioPlayer(self, didUpdateContext: context) }
+    }
+
+    // MARK: - Lifecycle
+
+    public static func build() -> AudioPlaying {
+        StreamRemoteAudioPlayer(
+            assetPropertyLoader: StreamAssetPropertyLoader(),
+            playerObserver: StreamPlayerObserver(),
+            player: .init()
+        )
+    }
+
+    public init(
+        assetPropertyLoader: AssetPropertyLoading,
+        playerObserver: AudioPlayerObserving,
+        player: AVPlayer
+    ) {
+        self.assetPropertyLoader = assetPropertyLoader
+        self.playerObserver = playerObserver
+        self.player = player
+
+        setUp()
+    }
+
+    // MARK: - Helpers
+
+    private func setUp() {
+        let player = self.player
+        let interval = CMTime(
+            seconds: 0.1,
+            preferredTimescale: CMTimeScale(NSEC_PER_SEC)
+        )
+
+        playerObserver.addPeriodicTimeObserver(
+            player,
+            forInterval: interval,
+            queue: nil
+        ) { [weak self] in
+            guard
+                let self = self,
+                self.context.isSeeking == false
+            else {
```

Comment:
Following the current codebase, should be something like
```suggestion
            guard let self = self, self.context.isSeeking == false else {
```